src/mono/mono/mini/method-to-ir.c

```diff
@@ -7562,9 +7564,22 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 								++new_param_count;
 							}
 						} else {
-							// For structs that cannot be lowered, we change the argument to byref type
+							// For structs that cannot be lowered, we change the argument to a pointer-like argument type.
+							// If SwiftSelf<T> can't be lowered, it should be passed in the same manner as SwiftSelf, via the context register.
+							if (gklass && (gklass->container_class == swift_self_t))
+								ptype = mono_class_get_byref_type (swift_self);
+							else
+								ptype = mono_class_get_byref_type (klass);
+
+							// Load the address of the struct with 0 offset from SwiftSelf<T>
+							if (gklass && (gklass->container_class == swift_self_t))
+							{
+								MonoInst *swift_self_base_address = struct_base_address;
+								struct_base_address = mono_compile_create_var (cfg, mono_get_int_type (), OP_LOCAL);
+								MONO_EMIT_NEW_BIALU_IMM (cfg, OP_ADD_IMM, struct_base_address->dreg, swift_self_base_address->dreg, 0);
```

Comment:
After all these discussions, it raises the trivial question. What is the point of this then ? We are just adding 0 to an address :)

Reply:
The instruction was not necessary and was converting `SwiftSelf<T>` to `IntPtr`. Using `SwiftSelf<T>` in the `ARGLOADA` was resulting in an assertion in the IR at some point. I've removed this instruction and updated the `ARGLOADA` instruction to be a pointer-like type. Please let me know if you have any other suggestions.